misc/ls-extensions/modules/json-to-record-converter/src/main/java/io/ballerina/converters/JsonToRecordConverter.java

```diff
@@ -368,6 +371,23 @@ private static TypeDescriptorNode extractOpenApiSchema(Schema<?> schema, String
         Token typeName = AbstractNodeFactory.createToken(SyntaxKind.ANY_KEYWORD);
         return createBuiltinSimpleNameReferenceNode(null, typeName);
     }
+    
+    private static ArrayTypeDescriptorNode createArrayTypeDesc(TypeDescriptorNode memberTypeDesc, 
+                                                               Token openBracketToken, Node arrayLengthNode, 
+                                                               Token closeBracketToken) {
+        NodeList<ArrayDimensionNode> arrayDimensions = NodeFactory.createEmptyNodeList();
+        if (memberTypeDesc.kind() == SyntaxKind.ARRAY_TYPE_DESC) {
+            ArrayTypeDescriptorNode innerArrayType = (ArrayTypeDescriptorNode) memberTypeDesc;
+            arrayDimensions = innerArrayType.dimensions();
+            memberTypeDesc = innerArrayType.memberTypeDesc();
+        }
+
+        ArrayDimensionNode arrayDimension = NodeFactory.createArrayDimensionNode(openBracketToken, arrayLengthNode,
+                closeBracketToken);
+        arrayDimensions.add(arrayDimension);
```

Comment:
Here `arrayDimensions.add()` gives a new NodeList since Syntax tree is immutable. May it was the reason for the build failure as well.

```suggestion
        arrayDimensions = arrayDimensions.add(arrayDimension);
```

Reply:
Tried with change locally. Tests are passing.

Reply:
Updated thanx